src/plugins/auto_batch/auto_batch.cpp

```diff
@@ -577,6 +577,8 @@ InferenceEngine::IInferRequestInternal::Ptr AutoBatchExecutableNetwork::CreateIn
     if (!_network) {
         auto res = _networkWithoutBatch->CreateInferRequest();
         res->setPointerToExecutableNetworkInternal(shared_from_this());
+        res->setPointerToSo(_networkWithoutBatch._so);
+        _so = _networkWithoutBatch._so;
```

Comment:
why the code below `if` does not set `_so`?

Reply:
per my understanding for batch,  only the code in this if is passing hw infer requests directly to application, which may expose the hw blobs directly to user. in other situation, batch plugin is allocating its own buffer.